public/app/features/dashboard/components/ShareModal/utils.ts

```diff
@@ -181,4 +181,80 @@ export const shareDashboardType: {
   report: 'report',
   publicDashboard: 'public_dashboard',
   inviteUser: 'invite_user',
+  image: 'image',
 };
+
+interface DashboardPanel {
+  gridPos?: {
+    y: number;
+    h: number;
+  };
+}
+
+interface Dashboard {
+  panels: DashboardPanel[];
+}
+
+function calculateDashboardHeight(dashboard: Dashboard): number {
```

Comment:
this piece I am the least sure about, also if width is always going to be ok or if we also need to calculate max width

this solution also wont work longer term for future non-grid dynamic dashboard layouts so we may need to rethink it at that time